[storybook] Removed unused addons

EvanBacon opened the pull request:
# How

- Removed extra styles from test snapshots
- Removed markdown-only tests from snapshots
- Made CI logs clearer
- Removed all warnings
* Removed notes
* Removed actions
* Removed events
* Default empty docs to MD
* Moved AV to new category
* Moved Sensors to new category
* Organized order of categories

# Test Plan

CI should continue to pass.

EvanBacon committed:
Removed notes

* Removed actions
* Removed events
* Default empty docs to MD
* Moved AV to new category
* Moved Sensors to new category
* Organized order of categories

EvanBacon committed:
updated tests

EvanBacon committed:
restructure tests

ide commented on apps/storybook/.storybook/config.js:
```diff
@@ -68,64 +68,63 @@ addParameters({
+
+  let markdown = notes;
+  if (!notes) {
+    const mdPath = filename.substr(0, filename.lastIndexOf('.stories')) + '.notes.md';
```
substr -> substring (former is deprecated-ish)

EvanBacon committed:
Remove image tests

EvanBacon committed:
Merge branch 'master' into @evanbacon/storybook/slim-code

EvanBacon committed:
Reverted image snapshots for PR

EvanBacon committed:
Fixed HMR

- Removed padded warnings

EvanBacon committed:
Fixed unmount error

EvanBacon committed:
Removed extra styles from test snapshots

- Removed markdown-only tests from snapshots
- Made CI logs clearer
- Removed all warnings

EvanBacon committed:
Fixed CI build messages

EvanBacon committed:
Optimized assets

EvanBacon committed:
Update image-storyshots.__test__.js

ide commented on apps/storybook/.storybook/image-storyshots.__test__.js:
```diff
@@ -1,16 +1,12 @@
 if (process.env.BABEL_ENV === 'test:web' && !process.env.IS_EXPO_CI) {
   // Locally it's better to start the server and then run the tests.
   // In CI we want to build a static bundle and test against that.
+  // const storybookUrl = 'file://' + path.resolve(__dirname, '..', 'web-build')
```
remove this

EvanBacon replied:
It works locally but breaks if used in CI. WIP

ide commented on apps/storybook/stories/APIs/Accelerometer.stories.jsx:
```diff
@@ -42,7 +46,9 @@ export class component extends React.Component {
   _subscribe = () => {
     this._subscription = Accelerometer.addListener(accelerometerData => {
-      this.setState({ accelerometerData });
+      if (this._isMounted) {
```
It seems like this covers up a bug -- the listener should no longer run once `componentWillUnmount` runs. Figure out why the listener is still being run and fix the issue at the source instead of patching over it.

ide commented on apps/storybook/stories/ui-explorer/SyntaxHighlighter.js:
```diff
@@ -209,24 +217,22 @@ export default class CopyableCode extends Component {
-    this.setState({ copied: true }, () => {
-      window.setTimeout(() => this.setState({ copied: false }), 1500);
-    });
+    if (this._isMounted) {
```
You shouldn't need this check -- onClick will run only when the component is mounted if the code is written correctly.

EvanBacon replied:
This was the one that was actually throwing

ide reviewed (changes requested):
Looks mostly good, but the "is mounted" checks need to be revised (in a separate PR is fine if you'd like to land the rest).

EvanBacon committed:
Removed extra mounting checks

EvanBacon committed:
Merge branch 'master' into @evanbacon/storybook/slim-code

EvanBacon committed:
Merge branch 'master' into @evanbacon/storybook/slim-code
